src/reference/03-Developers-Guide/00.md

```diff
@@ -2,12 +2,93 @@
 ---
 
-Developer's Guide
------------------
-
-This is the set of documentation about the Architecture of sbt. This
-covers all the core components of sbt as well as the general notion of
-how they all work together. This documentation is suitable for those who
-wish to have a deeper understanding of sbt's core, but already
-understand the fundamentals of `Setting[_]`, `Task[_]` and constructing
-builds.
+  [pickling]: https://github.com/scala/pickling
+  [HowCanIHelp]: http://www.scala-sbt.org/community.html#how-can-I-help
+  [Sbt-Launcher]: Sbt-Launcher.html
+
+Developer's Guide (Work in progress)
+------------------------------------
+
+This is the set of documentation about the Architecture of sbt.
+The target audience of this document is the sbt plugin authors
+and sbt developers. See also [How can I help?][HowCanIHelp]
+
+### Modularization
+
+The process we aim to take for sbt 1.0 is to disassemble sbt into small modules.
+To be clear sbt 0.13's codebase already does consists of numerous subprojects.
+
+Here's what we envision the modilarization process to look like:
+
+- pull apart subprojects by feature
+- make separate Github repositories
+- distinguish public API and internal implemention
+- document the usages
+- clean up the code since we can break bincompat
+- publish to Maven
+
+Here are some of the modules.
+
+#### [sbt/serialization](https://github.com/sbt/serialization)
+
+sbt serialization is an opinionated wrapper around [Scala Pickling][pickling] focused on sbt's usage.
+
+#### [launcher][Sbt-Launcher]
+
+The sbt launcher provides a generic container that can load and run
+programs resolved using the Ivy dependency manager. Sbt uses this as its
+own deployment mechanism. (Hosted at [sbt/launcher](https://github.com/sbt/launcher))
+
+#### [sbt/website](https://github.com/sbt/website)
+
+This website's source.
+
+#### [sbt/sbt](https://github.com/sbt/sbt)
+
+Most of the sbt codebase still lives here.
+
+### Towards sbt 1.0
+
+On 2008-12-18, Mark Harrah announced sbt 0.3.2 as the initial release of sbt.
+Mark remained the primary author of sbt until sbt 0.13.1 (2013-12-11).
+In 2014, sbt project was handed over to the authors of this document Josh Suereth
+and Eugene Yokota.
+
+As we move towards sbt 1.0, we wish to stablize what's already stable
+and innovate where it matters. There are several levels of stability:
+
+- conceptual stability
+- source compatibility of the build defnition
+- binary compatibility of the plugins
+
+#### Concepts
+
+Concepturally we are stable on what sbt does:
+
+1. incremental compilation that supports Scala
+2. dependency management that's aware of Scala's binary compatibility
+3. task and plugins system that's extensible using Scala
+4. a text-based interactive shell
+
+The only thing that will change is the last point.
+In sbt 1.0, we will replace the interactive shell with sbt server
+that's accessible via JSON API and a text-based client.
+
+#### Source compatibility of the build deinition
```

Comment:
Add a notice on "compatibility".   i.e. that we aim to keep compatibility between releases.  Some kind of mantra:

"Never break plugins, Never break build source files."